script.js

```diff
@@ -1,4 +1,205 @@
+- Player to choose Dice roll 1 or Dice roll 2 as the first digit
+- Transform the number into string, and concatenate them 
+- Store into final number array
+- Initiate dice roll mode for player 2
+
+Player 2 step
+- Let player 2 roll the 2 dice
+- Store the 2 dice numbers in the player dice choose array
+- Once 2 dice are rolled, initiate choose dice order game mode
+- Player to choose Dice roll 1 or Dice roll 2 as the first digit
+- Transform the number into string, and concatenate them 
+- Store into final number array, after player 1's number
+
+Comparison step
+- Compare the 2 numbers
+- Show in message box who is the winner
+
+Reset step
+- Reset game to initiate dice roll mode, upon declaration of the winner
+
+*/
+
+//global variables
+var diceRollGameModePlayer1 = "Initiate dice roll for player 1";
+var diceOrderGameModePlayer1 = "Choose dice order for player 1";
+var diceRollGameModePlayer2 = "Initiate dice roll for player 2";
+var diceOrderGameModePlayer2 = "Choose dice order for player 2";
+var compareScoreGameMode = "compare score";
+var gameMode = diceRollGameModePlayer1;
+var diceRollArray1 = [];
+var diceRollArray2 = [];
+var chooseNumberArray = [];
+var doubleDigitNumber = ``;
+var counter = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == diceRollGameModePlayer1) {
+    // define variable for PLayer 1's roll dice array
+    var myOutputValue = rollAndStoreDicePlayer1();
+
+    //change game mode to dice order game mode
+    gameMode = diceOrderGameModePlayer1;
+    console.log(gameMode);
+    return myOutputValue;
+  }
+
+  // roll dice and set order for Player 1
+  if (gameMode == diceOrderGameModePlayer1) {
+    if (input == 1) {
+      doubleDigitNumber = parseInt(`${diceRollArray1[0]}${diceRollArray1[1]}`);
+      chooseNumberArray.push(doubleDigitNumber);
+
+      gameMode = diceRollGameModePlayer2;
+      return `Hi Player 1, you have chosen ${doubleDigitNumber}. Now it is Player 2's turn to play.`;
+    }
+    if (input == 2) {
+      doubleDigitNumber = parseInt(`${diceRollArray1[1]}${diceRollArray1[0]}`);
+      chooseNumberArray.push(doubleDigitNumber);
+
+      gameMode = diceRollGameModePlayer2;
+
+      return `Hi Player 1, you have chosen ${doubleDigitNumber}. Now it is Player 2's turn to play.`;
+    }
+    if (input != 1 || input != 2) {
+      return `Hi Player 1, you have not typed in 1 or 2, please type in 1 or 2 <br> <br>
+      You have rolled Dice 1 as ${diceRollArray1[0]} and Dice 2 as ${diceRollArray1[1]}. <br> <br>`;
+    }
+  }
+
+  if (gameMode == diceRollGameModePlayer2) {
+    // define variable for PLayer 2's roll dice array
+    var myOutputValue2 = rollAndStoreDicePlayer2();
+
+    //change game mode to dice order game mode
+    gameMode = diceOrderGameModePlayer2;
+
+    return myOutputValue2;
+  }
+
+  // roll dice and set order for Player 2
+  if (gameMode == diceOrderGameModePlayer2) {
```

Comment:
I'm sure you have realized that line 116 - 140 is almost exactly the same as line 83 -101. Can you think of a way to refactor your code so that you dont have to repeat all these lines of code again in lines 116 - 140? 

You can actually refactor lines 83-101 into a helper function, then replace lines 83-101 and lines 116-140 with a single line where you call this helper function. 

Your helper function can look like this:

````
var chooseOrder = function (input, diceRollArray) {
    if (input == 1) {
      doubleDigitNumber = parseInt(`${diceRollArray[0]}${diceRollArray[1]}`);
      chooseNumberArray.push(doubleDigitNumber);

      // move to compare score game mode
      gameMode = compareScoreGameMode;
      return `Hi Player 2, you have chosen ${doubleDigitNumber}.`;
    }
    if (input == 2) {
      doubleDigitNumber = parseInt(`${diceRollArray[1]}${diceRollArray[0]}`);
      chooseNumberArray.push(doubleDigitNumber);

      // move to compare score game mode
      gameMode = compareScoreGameMode;
      return `Hi Player 2, you have chosen ${doubleDigitNumber}.`;
    }
    if (input != 1 || input != 2) {
      return `Hi Player 2, you have not typed in 1 or 2, please type in 1 or 2 <br> <br>
      You have rolled Dice 1 as ${diceRollArray[0]} and Dice 2 as ${diceRollArray[1]}. <br> <br>`;
    }
}
````
Then when you call this function, simply assign the return value to a variable then return the variable i.e.
````
var outputValue = chooseOrder(input, diceRollArray1/diceRollArray2)
return outputValue
````